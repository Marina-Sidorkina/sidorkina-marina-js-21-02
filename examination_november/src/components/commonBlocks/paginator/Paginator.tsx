import React, { useContext } from 'react';
import './Paginator.scss';
import { Pagination } from 'antd';
import { ThemeContext } from '../../../contexts/ThemeContext';

interface IPaginatorProps {
  perPage: number | undefined;
  current: number;
  total: number;
  onPageChange: (page: number) => void;
  modal?: boolean;
}

const Paginator = (props: IPaginatorProps) => {
  const {
    perPage, onPageChange, current, total, modal
  } = props;
  const themeContext = useContext(ThemeContext);

  return (
    <Pagination
      className={
        `${modal
          ? 'paginator paginator_modal'
          : 'paginator'} ${themeContext.darkTheme ? 'paginator_dark' : ''}`
      }
      simple
      pageSize={perPage}
      total={total}
      current={current}
      onChange={onPageChange}
    />
  );
};

Paginator.defaultProps = {
  modal: false
};

export default Paginator;
