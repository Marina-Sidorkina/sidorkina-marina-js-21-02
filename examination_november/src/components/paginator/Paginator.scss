.paginator {
  position: absolute;
  bottom: 15px;
  right: 15px;

  &_modal {
    position: static;
  }
}