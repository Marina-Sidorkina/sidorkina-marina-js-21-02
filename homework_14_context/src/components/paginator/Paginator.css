.paginator {
  display: flex;
  justify-content: flex-start;
  padding: 0;
  margin: 0;
  list-style: none;
}

.paginator__item {
  width: 33px;
  height: 42px;
  margin-right: 4px;
  font-size: 28px;
  background-color: white;
  border-radius: 5px;
}

.paginator_dark .paginator__link {
  color: white;
  background-color: #0d1117;
  border: 1px solid white;
}

.paginator_dark .paginator__link_current {
  color: white;
  background-color: #6c7c93;
}

.paginator__link {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 33px;
  height: 42px;
  color: black;
  text-decoration: none;
  border: 1px solid black;
  border-radius: 5px;
}

.paginator__link:hover {
  background-color: #e6f2ff;
}

.paginator__link:active {
  color: white;
  background-color: #b3d9ff;
}

.paginator__link_current {
  color: white;
  background-color: #b3d9ff;
}
