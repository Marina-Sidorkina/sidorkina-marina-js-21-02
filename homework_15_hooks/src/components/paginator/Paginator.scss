.paginator {
  display: flex;
  justify-content: flex-start;
  padding: 0;
  margin: 0;
  list-style: none;

  &__item {
    width: 33px;
    height: 42px;
    margin-right: 4px;
    font-size: 28px;
    background-color: white;
    border-radius: 5px;
  }

  &_dark .paginator__link {
    color: white;
    background-color: #0d1117;
    border: 1px solid white;
  }

  &_dark .paginator__link_current {
    color: white;
    background-color: #6c7c93;
  }

  &__link {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 33px;
    height: 42px;
    color: black;
    text-decoration: none;
    border: 1px solid black;
    border-radius: 5px;

    &:hover {
      background-color: #e6f2ff;
    }

    &:active {
      color: white;
      background-color: #b3d9ff;
    }

    &_current {
      color: white;
      background-color: #b3d9ff;
    }
  }
}
