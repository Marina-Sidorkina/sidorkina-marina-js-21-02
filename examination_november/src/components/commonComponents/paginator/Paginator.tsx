import React, { useContext } from 'react';
import './Paginator.scss';
import { Pagination } from 'antd';
import { ThemeContext } from '../../../contexts/ThemeContext';
import { IPaginatorProps } from './@types/paginator';

const Paginator = (props: IPaginatorProps) => {
  const {
    perPage, onPageChange, current, total, modal
  } = props;
  const themeContext = useContext(ThemeContext);

  return (
    <Pagination
      className={
        `${modal
          ? 'paginator paginator_modal'
          : 'paginator'} ${themeContext.darkTheme ? 'paginator_dark' : ''}`
      }
      simple
      pageSize={perPage}
      total={total}
      current={current}
      onChange={onPageChange}
    />
  );
};

Paginator.defaultProps = {
  modal: false
};

export default Paginator;
