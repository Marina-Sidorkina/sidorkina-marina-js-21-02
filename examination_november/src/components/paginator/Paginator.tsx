import React from 'react';
import './Paginator.scss';
import { Pagination } from 'antd';

interface IPaginatorProps {
  perPage: number | undefined;
  current: number;
  total: number;
  onPageChange: (page: number) => void;
  modal?: boolean;
}

const Paginator = (props: IPaginatorProps) => (
  <Pagination
    className={props.modal ? 'paginator paginator_modal' : 'paginator'}
    simple
    defaultPageSize={props.perPage}
    defaultCurrent={props.current}
    total={props.total}
    onChange={props.onPageChange}
  />
);

Paginator.defaultProps = {
  modal: false
};

export default Paginator;
