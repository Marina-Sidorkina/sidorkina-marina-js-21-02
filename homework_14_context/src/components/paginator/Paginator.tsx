import React, {SyntheticEvent} from "react";
import "./Paginator.css";
import { IPaginatorProps } from "../../@types/interfaces/components";

class Paginator extends React.Component<IPaginatorProps> {
  constructor(props: IPaginatorProps) {
    super(props);

    this.onCurrentChange = this.onCurrentChange.bind(this);
  }

  onCurrentChange(evt: SyntheticEvent) {
    const value = parseInt((evt.target as HTMLElement).innerText, 10);
    this.props.onCurrentChange(value);
  }

  render() {
    const elements = [...Array(5)].map((_, index) => (
      <li className="paginator__item" key={ index } onClick={ this.onCurrentChange }>
        <a className={`paginator__link ${ this.props.current === index + 1 ? 
                        "paginator__link_current" : ""}`}
           href="#">{ index + 1 }</a>
      </li>
    ));

    return (
      <ul className={ `paginator ${ this.props.className }` }>
        { elements }
      </ul>
    );
  }
}

export default Paginator;
