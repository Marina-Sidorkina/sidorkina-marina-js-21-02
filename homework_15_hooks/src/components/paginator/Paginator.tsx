import React, {SyntheticEvent, useContext} from "react";
import "./Paginator.scss";
import { IPaginatorProps } from "../../@types/interfaces/components";
import { ThemeContext } from "../../contexts/ThemeContext";

const Paginator = (props: IPaginatorProps) => {
  const themeContext = useContext(ThemeContext);

  const onCurrentChange = (evt: SyntheticEvent) => {
    const value = parseInt((evt.target as HTMLElement).innerText, 10);
    props.onCurrentChange(value);
  }

  const elements = [...Array(5)].map((_, index) => (
    <li className="paginator__item" key={ index } onClick={ onCurrentChange }>
      <a className={`paginator__link ${ props.current === index + 1 ?
        "paginator__link_current" : ""}`}
         href={ `#${ index + 1 }` }>{ index + 1 }</a>
    </li>
  ));

  return (
    <ul className={ `paginator ${ themeContext.darkTheme ? "paginator_dark" : "" }` }>
      { elements }
    </ul>
  );
}

export default Paginator;
