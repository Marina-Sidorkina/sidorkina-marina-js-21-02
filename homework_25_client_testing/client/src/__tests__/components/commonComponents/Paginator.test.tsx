import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Paginator from '../../../components/commonComponents/paginator/Paginator';

configure({ adapter: new Adapter() });

describe('Paginator component test', () => {
  test('should render Paginator', () => {
    const wrapper = shallow(<Paginator
      perPage={5}
      onPageChange={() => {}}
      current={0}
      total={10}
    />);
    expect(wrapper.length).toEqual(1);
  });

  test('should render Paginator', () => {
    const wrapper = shallow(<Paginator
      perPage={5}
      onPageChange={() => {}}
      current={0}
      total={10}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
