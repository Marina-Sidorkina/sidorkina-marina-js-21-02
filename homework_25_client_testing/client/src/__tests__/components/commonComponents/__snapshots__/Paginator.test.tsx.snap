// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component test should render Paginator 1`] = `
<Pagination
  className="paginator
        "
  current={0}
  onChange={[Function]}
  pageSize={5}
  showQuickJumper={true}
  simple={true}
  total={10}
/>
`;
